### PR TITLE
Omitting deprecated `"users.*"` settings for TiFlash

### PR DESCRIPTION
**Pull request by JaySon-Huang**
Signed-off-by: JaySon-Huang <tshent@qq.com>

### What problem does this PR solve? 

A part of https://github.com/pingcap/tics/issues/1497.
TiFlash PRs on 5.0.x and 4.0.x: https://github.com/pingcap/tics/pull/1526, https://github.com/pingcap/tics/pull/1535

Some access control settings are inherited from Clickhouse and useless for TiFlash, we make these settings to be the default value if empty since v4.0.12 and v5.0.0. Some settings (like `password = ""`) make trouble for TiFlash users when their configuration files are scanned by a security scanner. 

This PR make TiUP don't generate those deprecated settings for cluster version >= v4.0.12 and not euqals to v5.0.0-rc

### What is changed and how it works?

Ignore the `'users.*'` and `'quotas.*'` settings in TiFlash if the cluster version >= v4.0.12 and not equals to v5.0.0-rc.
But keep those settings for version <= v4.0.11

### Check List 

Tests 

 - Unit test

Code changes

 - Has persistent data change

Side effects

 - N/A

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

Release notes:

```release-note
Omitting deprecated `"users.*"` settings in the configuration file of TiFlash.
```

### CODE REVIEW

**Comment by JaySon-Huang** [comment]
Do not merge until we confirm that v4.0.12 will contain that commit.

**Comment by JaySon-Huang** [comment]
@AstroProfundis PTAL

File: pkg/cluster/spec/tiflash.go [hunk 1]
```diff
@@ -367,6 +368,32 @@ func (i *TiFlashInstance) initTiFlashConfig(cfg *scripts.TiFlashScript, clusterV
 		pathConfig = fmt.Sprintf(`path: "%s"`, cfg.DataDir)
 	}
 
+	if (semver.Compare(clusterVersion, "v4.0.12") >= 0 && semver.Compare(clusterVersion, "v5.0.0-rc") != 0) || clusterVersion == "nightly" {
```
Comment (AstroProfundis) [inline pkg/cluster/spec/tiflash.go:371]: It's better to use `utils.Version(clusterVersion).IsNightly()` to check for nightly versions, the actual version in topo will be expanded from `nightly` to a specific version at that date.

Comment (JaySon-Huang) [reply pkg/cluster/spec/tiflash.go]: I push a commit to use `utils.Version(clusterVersion).IsNightly()` for TiFlash.

But I'm wondering should I also change this code in `playground.go`?
https://github.com/pingcap/tiup/blob/bd3cf97ef84f62be369b1c4af95b5901872c2803/components/playground/playground.go#L688

Comment (lucklove) [reply pkg/cluster/spec/tiflash.go]: I think so, it's better to change playground.go too.

Comment (JaySon-Huang) [reply pkg/cluster/spec/tiflash.go]: Addressed.

**Review by AstroProfundis — APPROVED** [review]

**Pull request merged**